src/System.Reflection.Metadata/src/System/Reflection/Metadata/Handles.cs

```diff
@@ -7,77 +7,179 @@
 
 namespace System.Reflection.Metadata
 {
+    /// <summary>
+    /// Represents any metadata entity (type reference/definition/specification, method definition, custom attribute, etc.) or value (string, blob, guid, user string).
+    /// </summary>
+    /// <remarks>
+    /// Use <see cref="Handle"/> to store multiple kinds of handles.
+    /// </remarks>
     public struct Handle : IEquatable<Handle>
     {
-        internal readonly uint value;
+        private readonly int _value;
 
-        internal Handle(uint value)
+        // bits:
+        //    7: IsVirtual
+        // 0..6: token type
+        private readonly byte _vType;
```

Comment:
I have serious concerns about the performance implications of adding this. I haven't convinced myself that it is actually necessary. To help with this concern, can you answer the following two questions?

* :question: For which value(s) of `_vType` does `_value` have a 29-bit supported range?
* :question: For which value(s) of `_vType` does `_value` have less than 29-bit supported range? In these cases, what is the supported range for `_value`?